src/aws/flb_aws_util.c

```diff
@@ -705,7 +707,7 @@ flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag,
     tmp = NULL;
 
     /* Split the string on the delimiters */
-    tag_token = strtok(tmp_tag, tag_delimiter);
+    tag_token = strtok_r(tmp_tag, tag_delimiter, &strtok_saveptr);
```

Comment:
please get rid of strtokX() usage. use https://github.com/fluent/fluent-bit/blob/master/src/flb_utils.c#L255 instead

Reply:
@edsiper  Unfortunately we can not do that right now. `flb_utils_split` only splits on a single separate and we used `strtok` here because it supports multiple. So I can not refactor this to implement your suggestion. This fix is needed for a bug. Can we merge it and then consider refactoring long term?

Reply:
got it, let's make sure to get rid of strtok() for Fluent Bit v2.0

Reply:
Discussed via Slack and agreed to use strtok for now to fix 1.x and I will use an ifdef in our code to use this on windows: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/strtok-s-strtok-s-l-wcstok-s-wcstok-s-l-mbstok-s-mbstok-s-l?view=msvc-170